src/main/java/de/cerus/jdasc/JDASlashCommands.java

```diff
@@ -442,25 +530,29 @@ private static InteractionResponseOption findArgument(final ApplicationCommand c
         walkList(rspOptions, interaction.getOptions(), option -> option.getOptions() != null && option.getOptions().size() > 0, InteractionResponseOption::getOptions);
         return rspOptions.stream()
                 .filter(option -> argNames.contains(option.getName()))
-                .findAny()
-                .orElse(null);
+                .collect(Collectors.toMap(InteractionResponseOption::getName, interactionResponseOption -> interactionResponseOption));
     }
 
+
     private static <T> void walkList(final List<T> output, final List<T> list, final Predicate<T> predicate, final Function<T, List<T>> function) {
-        for (final T item : list) {
-            if (predicate.test(item)) {
-                walkList(output, function.apply(item), predicate, function);
-            } else {
-                output.add(item);
+        //Yes this is a bad practice, but idk whats null lol
+        try {
+            for (final T item : list) {
+                if (predicate.test(item)) {
+                    walkList(output, function.apply(item), predicate, function);
+                } else {
+                    output.add(item);
+                }
             }
+        } catch (NullPointerException ignored) {
```

Comment:
I'd like to keep this project as clean as possible so I'd really appreciate if you could take a look at this again.

Reply:
Lmk if this is what you ment